larg4/pluginActions/ParticleListAction_service.cxx

```diff
@@ -537,6 +585,14 @@ namespace larg4 {
 // event and pass the call on to the action objects.
   void ParticleListActionService::endOfEventAction(const G4Event*)
 {
+  // -- End of Run Report
+  std::stringstream sscounter;
+  sscounter << "Not Stored Process summary:";
+  for( auto c : fNotStoredCounterUMap ){
+    sscounter << "\n\t" << c.first << " : " << c.second;
```

Comment:
Two issues: 

- `auto c` is a copy of each pair in the map, which can be expensive.  It would be better to use `auto const& c`, which iterates using a reference instead.
- C++17 allows you to use structured bindings so that you can write this instead:

```cpp
for (auto const& [process, count] : fNotStoredCounterUMap) {
  sscounter << "\n\t" << process << " : " << count;
}
```

Notice the `auto const&` in the `for` loop, which avoids the copy.

Reply:
This is great, thank you.